Sources/swift-openapi-generator/GenerateOptions.swift

```
@@ -86,7 +86,42 @@ extension _GenerateOptions {
         if !featureFlag.isEmpty {
             return Set(featureFlag)
         }
-        return Set(config?.featureFlags ?? [])
+        return config?.featureFlags ?? []
+    }
+
+    /// Extracts the top-level keys from a YAML string.
+    ///
+    /// - Parameter yamlString: The YAML string from which to extract keys.
+    /// - Returns: An array of top-level keys as strings.
+    func extractTopLevelKeys(fromYAMLString yamlString: String) -> [String] {
+        var yamlKeys = [String]()
+
+        do {
+            let parser = try Parser(yaml: yamlString)
+            if let rootNode = try parser.singleRoot(),
+                case let .mapping(mapping) = rootNode
+            {
+                for (key, _) in mapping {
+                    yamlKeys.append(key.string ?? "")
+                }
+            }
+        } catch {
+            print("Error parsing YAML: \(error)")
```

I think it's better to just let the error be thrown one level up, and there emit a ValidationError saying the config isn't valid and include the description of the error thrown by Yams. So you can remove the do/catch here.

---

Sure thing!!

---

Just a quick question, where would be a good place in the codebase, to write the unit test?

---

You could move the logic close to the YAML parser in the core module, and that one has unit tests that you can add to. We don't yet have a unit test target for the CLI module.

---

perfect, TY!!